### ISSUE TITLE
The pre-trained model on ImageNet

### ISSUE THREAD

**Issue by Feynben** [XzwHan/CARD#1]
Hi!
Thanks for your contribution! Could you provide the pre-trained model on ImageNet or give some more details on the training of ImageNet, Thanks!

**Comment by XzwHan** [comment]
Hi @Fayeben, thank you for your question! We've provided the checkpoints for download in the [README file](https://github.com/XzwHan/CARD#checkpoints).

You may find two files in the [ImageNet folder](https://drive.google.com/drive/u/5/folders/1gnCyBU3uZxgbJqg9aPe66VZxdT1AZMxa), each of which corresponds to a $\boldsymbol{\epsilon_{\theta}}$ network trained with a different pre-trained classifier $\boldsymbol{f_{\phi}}$: one with a test accuracy of $76.13\\%$, the other $80.30\\%$. Both $\boldsymbol{f_{\phi}}$ have a ResNet-50 architecture, whose weights are provided by [`torchvision`](https://pytorch.org/vision/stable/models/generated/torchvision.models.resnet50.html#torchvision.models.resnet50) (**ResNet50_Weights.IMAGENET1K_V1** and **ResNet50_Weights.IMAGENET1K_V2**; note that the Top-1 test accuracy of **IMAGENET1K_V2** has been updated, but when we were running our experiments, it was $80.30\\%$). You can find more details, including the training paradigms of these two versions of $\boldsymbol{f_{\phi}}$ network, in Appendix **A.7** of our paper.

You may also use our code to train the $\boldsymbol{\epsilon_{\theta}}$ network for the ImageNet classification task: we applied distributed training through PyTorch DDP to expedite the training, thus modified the original code from [`card_classification.py`](https://github.com/XzwHan/CARD/blob/main/classification/card_classification.py) file specifically for large-scale datasets. You can find the implementation in [`imagenet`](https://github.com/XzwHan/CARD/tree/main/classification/imagenet) directory, in which the [`main.py`](https://github.com/XzwHan/CARD/blob/main/classification/imagenet/main.py) file contains the code for training, and the [`builder.py`](https://github.com/XzwHan/CARD/blob/main/classification/imagenet/card/builder.py) file has the code of network architecture (`self.model_phi` and `self.model_theta` correspond to $\boldsymbol{f_{\phi}}$ and $\boldsymbol{\epsilon_{\theta}}$ network, respectively).

**Comment by Feynben** [comment]
Thanks for your patient reply! It helps a lot.
